src/platform_impl/linux/x11/mod.rs

```diff
@@ -1128,3 +1128,9 @@ impl Device {
         }
     }
 }
+
+/// Convert the raw X11 representation for a 32-bit floating point to a double.
+#[inline]
+fn xinput_fp1616_to_float(fp: xinput::Fp1616) -> f64 {
+    (fp as f64) / ((1 << 16) as f64)
```

Comment:
That looks like should be a part of x11rb?

Reply:
I don't know. `x11rb` mostly deals with fetching and retrieving data from the server and not interpreting it.

@psychon Thoughts on this?

Reply:
Either way it will probably be a while before the next `x11rb` release, so to fix this bug let's just include it here.

Reply:
> @psychon Thoughts on this?

"xcb doesn't provide something like this either", but that's no good reason. x11rb also provides extra helper functions on `GetPropertyReply` like `value8()`, so we could also add `Fp1616::as_f64()` (suggestions for a name welcome). The property code seems to be done in `generator/src/generator/special_cases.rs`. So... feel free to open a PR for adding this case. Seems relatively simple and straight-forward.